src/ContractForm.js

```diff
@@ -6,59 +6,58 @@ class ContractForm extends Component {
   constructor(props, context) {
     super(props);
 
-    this.handleInputChange = this.handleInputChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
 
+    this.sendArgs = props.sendArgs || {};
+
     this.contracts = context.drizzle.contracts;
     this.utils = context.drizzle.web3.utils;
 
     // Get the contract ABI
     const abi = this.contracts[this.props.contract].abi;
 
     this.inputs = [];
-    var initialState = {};
 
     // Iterate over abi for correct function.
     for (var i = 0; i < abi.length; i++) {
       if (abi[i].name === this.props.method) {
         this.inputs = abi[i].inputs;
 
+        // create references for all values, to be used in the form elements
         for (var j = 0; j < this.inputs.length; j++) {
-          initialState[this.inputs[j].name] = "";
+          this.inputs[j].ref = React.createRef();
         }
-
         break;
       }
     }
+  }
 
-    this.state = initialState;
+  setSendArgs(args) {
```

Comment:
You can use `setSendArgs` in the render function to change the send argument. I need that in my application, because I must set a `value` which is calculated from the form input fields.

In my code, I have something like:
```
    inputs.fee.current.value = mul(this.state.amount, this.state.unit);
    parent.setSendArgs({ value: inputs.fee.current.value });
```